plugins/performance-lab/includes/admin/plugin-activate-ajax.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Handles activation of Performance Features (Plugins) using AJAX.
+ */
+
+/* global perflabPluginActivateAjaxData */
+( function ( $ ) {
+	// @ts-ignore
+	const { i18n, a11y } = wp;
+	const { __, _x } = i18n;
+
+	/**
+	 * Adds a click event listener to the plugin activate buttons.
+	 *
+	 * @param {MouseEvent} event - The click event object that is triggered when the user clicks on the document.
+	 *
+	 * @return {Promise<void>} - The asynchronous function returns a promise.
+	 */
+	$( document ).on(
+		'click',
+		'.perflab-install-active-plugin',
```

Comment:
What if someone clicks this button while another request is currently in flight? Should it short-circuit so as to avoid duplicating requests?

Reply:
I think as the request could have been half finished we should not short-circuit the request. I think it will be better to just disable it until the request is finished. But as the click event is on the anchor tag we will have to add `pointer-events: none` CSS to the anchor link using JavaScript to disable it.

Reply:
The `pointer-events` style wouldn't address the issue of the button being activated by keyboard.

Reply:
As the click event is on the anchor tag and not button we will have to add some other work around for disabling on keyboard one is setting its `tabindex` attribute to `-1`.

Reply:
I tested this but if user is already focused then `tabindex` will not work. One other way is to add a attribute like `aria-disabled` and then check on `click` event if it has that attribute then return from the execution. Should I add `tabindex` and `aria-disabled` both?

Reply:
You could add the `disabled` attribute, but that wouldn't be good because it would lose the tab focus for users navigating with the keyboard. Instead of this, WordPress has style rules for when you add the `disabled` class to a button so that it will look disabled. Adding `aria-disabled` here too would be good. But then for how to detect whether request is currently in progress, you could just short-circuit the click handler if the button has the `updating-message` class, or if it has the `disabled` class, or if it has the `aria-disabled` attribute.